spec.bs

```diff
@@ -4315,6 +4332,61 @@ at the expense of some utility.
   </wpt>
 </div>
 
+<h3 id=credentialless-monkeypatch>Iframe credentialless</h3>
+
+The [[!IFRAME-CREDENTIALLESS]] specification defines a new object, the [=page credentialless
+nonce=]. At a high level, the [=fenced frame config instance/partition nonce=] serves the same
+purpose as the [=page credentialless nonce=] (partitioning storage and network). Howevever, it is
+scoped to each fenced frame, whereas the [=page credentialless nonce=] is scoped to the top-level
+document and shared by all descendent credentialless iframes. The [=fenced frame config instance/
+partition nonce=] is also used to revoke network access in fenced frames. Apply the following
+monkeypatches to the [[!IFRAME-CREDENTIALLESS]] spec.
+
+Add the following algorithm:
+
+<div algorithm>
+  To <dfn for="browsing context">compute the effective partition nonce</dfn> given a boolean 
+  |credentialless| and [=fenced frame config instance/partition nonce=]-or-null
+  |newFencedFrameNonce|:
+
+  1. If |credentialless| is true, return the associated [=browsing context=]'s [=top-level
+     browsing context=]'s [=page credentialless nonce=].
+
+  1. If |newFencedFrameNonce| is not null, return |newFencedFrameNonce|.
+
+  1. Let |instance| be the associated [=browsing context=]'s [=browsing context/fenced frame
+     config instance=].
+
+  1. Return |instance|'s [=fenced frame config instance/partition nonce=] if |instance| is not
+     null, and return null otherwise.
+</div>
+
+<div algorithm="process a navigate fetch">
+  Modify the step added to <a href=https://wicg.github.io/anonymous-iframe/#spec-navigation-partition-nonce>
+  process a navigate fetch</a> to read:
+
+  13. Let <var ignore>partitionNonce</var> be the result of [=browsing context/computing the
+      effective partition nonce=] on <var ignore>browsingContext</var> given <var ignore>
+      credentialless</var> and null.
+
+  Note: We pass null to [=browsing context/computing the effective partition nonce=] here because
+  we want navigation fetches originating from a fenced frame to use the same partition as the
+  fenced frame itself. After the navigation fetch completes and we initialize the document, we can
+  access the correct partition nonce via the [=fenced frame config instance=]'s
+  [=fenced frame config instance/partition nonce=].
+</div>
+
+<div algorithm="initialize the document object">
+  Modify the step added to <a href=https://wicg.github.io/anonymous-iframe/#spec-window-partition-nonce>
+  initialize the document object</a> to read:
+
+  6.9. Let <var ignore>partitionNonce</var> be the result of [=browsing context/computing the
+       effective partition nonce=] on <var ignore>browsingContext</var> given |navigationParams|'s
```

Comment:
To confirm, this is what eventually ends up being put into the new environment object when it's created, right? The changes made to the fetch patches (where null is unconditionally passed in for the fenced frame nonce when computing the effective partition nonce) will have no effect on the environment that's ultimately created for the document?

Reply:
Ack, then I agree that this should be fine.